source/Cosmos.Core/Processing/ProcessorScheduler.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using IL2CPU.API.Attribs;
+
+namespace Cosmos.Core.Processing
+{
+    public static unsafe class ProcessorScheduler
+    {
+        public static void Initialize()
+        {
+            var context = new ProcessContext.Context();
+            context.type = ProcessContext.Context_Type.PROCESS;
+            context.tid = ProcessContext.m_NextCID++;
+            context.name = "Boot";
+            context.esp = 0;
+            context.stacktop = 0;
+            context.state = ProcessContext.Thread_State.ALIVE;
+            context.arg = 0;
+            context.priority = 0;
+            context.age = 0;
+            context.parent = 0;
+            ProcessContext.m_ContextList = context;
+            ProcessContext.m_CurrentContext = context;
+
+            IOPort counter0 = new IOPort(0x40);
+            IOPort cmd = new IOPort(0x43);
```

Comment:
Find a way to use another port or something else than PIT, with this using PC Speaker won't be possible and will hang the processor scheduler

Reply:
maybe pit?

Reply:
maybe use PIT implemented in Cosmos, not IOPort PIT?

Reply:
@AsertCreator they are one in the same

Reply:
oh okay. im trying to implement my own multitasking, using PIT override

Reply:
maybe you can implement the nanosecond time in RTC, and use RTC to approximate? or maybe (pure speculation, as a junior dev who knows nothing about low level) reimplement PIT wait to tally up the number of times IRQ 0 has been triggered since the os was powered on, and use this to wait _x_ amount of time. in the case of the second option, it would be possible to implement a system where multiple things can use PIT wait at the same time. see [osdev wiki](https://wiki.osdev.org/PIT#PIT_Channel_0_Example_Code).

Reply:
RTC is too slow for scheduling and idk if we could make interrupts with that

Reply:
In fact RTC may not be a bad idea until we have HPET or local APIC timers, it indeed can generate IRQs. I'll try, I hope this won't mess up date time

Reply:
Does that function as a stopgap until APIC gets finished?